qa/sc_bootstrap.py

```diff
@@ -0,0 +1,121 @@
+        mc_block_json = response["result"]["block"]["mainchainBlocks"][0]
+        new_boxes = mc_block_json["sidechainRelatedAggregatedTransaction"]["newBoxes"]
+        print("Checking that each public key has a box assigned with a non-zero value.")
+        for key in keys:
+            target = None
+            for box in new_boxes:
+                if box["proposition"]["publicKey"] == key["publicKey"]:
+                    target = box
+                    box_value = box["value"]
+                    assert_true(box_value > 0, "Non positive value for box: {0} with public key: {1}".format(box["id"], key))
+                    assert_equal(100*100000000, box_value, "Unexpected value for box: {0} with public key: {1}".format(box["id"], key))
+                    break
+            assert_true(target is not None, "Box related to public key: {0} not found".format(key))
+
+        print("Checking mainchain block reference inclusion.")
+        mc_block_version = mc_block["version"]
+        mc_block_merkleroot = mc_block["merkleroot"]
+        mc_block_time = mc_block["time"]
+        mc_block_bits = mc_block["bits"]
+        mc_block_nonce = mc_block["nonce"]
+        mc_block_previousblockhash = mc_block["previousblockhash"]
+        sc_mc_block_version = mc_block_json["header"]["version"]
+        sc_mc_block_merkleroot = mc_block_json["header"]["hashMerkleRoot"]
+        sc_mc_block_time = mc_block_json["header"]["time"]
+        sc_mc_block_bits = mc_block_json["header"]["bits"]
+        sc_mc_block_nonce = mc_block_json["header"]["nonce"]
+        sc_mc_block_previousblockhash = mc_block_json["header"]["hashPrevBlock"]
+        assert_equal(mc_block_version, sc_mc_block_version)
+        assert_equal(mc_block_merkleroot, sc_mc_block_merkleroot)
+        assert_equal(mc_block_time, sc_mc_block_time)
+        assert_equal(mc_block_nonce, sc_mc_block_nonce)
+        assert_equal(mc_block_previousblockhash, sc_mc_block_previousblockhash)
+
+    def run_test(self):
+        # start 1 mc node
+        mc_nodes = self.nodes
+        print "Number of MC nodes: {0}".format(len(mc_nodes))
+        assert_equal(1, len(mc_nodes), "The number of MC nodes is grater than 1.")
+
+        number_of_sidechains = 1
+        number_of_accounts_per_sidechain = []
+        for i in range(number_of_sidechains):
+            number_of_accounts_per_sidechain.append(i+1)
+
+        # Generate information for bootstrapping sidechains
+        sc_nodes_info = self.bootstrap_sidechain(number_of_sidechains, number_of_accounts_per_sidechain, mc_nodes[0])
+        assert_equal(number_of_sidechains, len(sc_nodes_info), "Not all sidechains have been successfully created.")
+        print sc_nodes_info
+
+        # Start sidechain nodes
+        sc_nodes = self.sc_setup_nodes(number_of_sidechains)
+
+        # Check validity of genesis information
+        for i in range(number_of_sidechains):
```

Comment:
only this part (from line 111 and below) should be present in this test.

Reply:
ok

Reply:
Done